### PR TITLE
Fix tar.gz support

### LINKED ISSUE #285: http fails to unpack certain tar.gz files
**What steps did you take:**

I wanted to vendor a tar.gz file fetched from github:

```
apiVersion: vendir.k14s.io/v1alpha1
kind: Config
directories:
- path: sponnet
  contents:
  - path: .
    http:
      url: https://github.com/spinnaker/sponnet/archive/refs/tags/v1.26.0.tar.gz
      unpackArchive: true
```

**What happened:**
The tar.gz file was not unpacked.

**What did you expect:**
The tar.gz file should have been unpacked.

**Anything else you would like to add:**
Originally I experimented with a fix for the issue mentioned in #150.

**Environment:**

```
❯ vendir version
vendir version 0.34.3

Succeeded
```
OS: MacOS Ventura 13.5

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by reegnz**
Fixes #285

### CODE REVIEW

**Comment by reegnz** [comment]
I haven't tried with other tar.gz files, but it definitely fixed the issue for this particular one of downloading a tar.gz of a github repo. Also the fix was inspired by this code in the go codebase: https://cs.opensource.google/go/x/build/+/2c66d68b:internal/untar/untar.go;l=34-139, particularly line 132 and 133.

**Comment by neil-hickey** [comment]
hey @reegnz , so this does appear to work on the surface. I think to really validate we are going to want some more e2e tests that cover each use-case. Is this something you would be able to contribute?

**Comment by reegnz** [comment]
@neil-hickey If you can point me to what exactly you'd like to see I can give it a go.

**Comment by joaopapereira** [comment]
I think we need something similar to https://github.com/carvel-dev/vendir/blob/develop/test/e2e/github_release_test.go but for HTTP workflow. We might need to standup a simple HTTP server and serve a tar.gz file from it to test it out.

**Comment by reegnz** [comment]
I think I'll just fetch a tar.gz from github (from this repo), that's one of those tar.gz-s what's causing the issue anyway, and that's the most life-like test.

**Comment by reegnz** [comment]
Pushed the e2e test (already had this in the making today, before the local http server idea).

File: test/e2e/http_tar_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestHttpTarGz(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	vendir := Vendir{t, env.BinaryPath, logger}
+	dstPath, err := os.MkdirTemp("", "vendir-e2e-http-targz-dst")
+	require.NoError(t, err)
+	// defer os.RemoveAll(dstPath)
```
Comment (joaopapereira) [inline test/e2e/http_tar_test.go]:
```suggestion
	defer os.RemoveAll(dstPath)
```

**Review by joaopapereira — CHANGES_REQUESTED** [review]
I left a comment. The linter is also failing do you mind giving it a check?

**Comment by reegnz** [comment]
@joaopapereira fixed.

**Comment by joaopapereira** [comment]
@reegnz you need to sign your commits 😄

**Comment by reegnz** [comment]
@joaopapereira rebased, added signoff to commits.

**Review by joaopapereira — APPROVED** [review]
LGTM

**Comment by kumaritanushree** [comment]
LGTM

**Pull request merged**